core/src/test/java/org/bitcoinj/core/PeerTest.java

```diff
@@ -478,22 +478,22 @@ public void fastCatchup() throws Exception {
         peer.setDownloadParameters(Utils.currentTimeSeconds() - (600*2) + 1, false);
         peer.startBlockChainDownload();
         GetHeadersMessage getheaders = (GetHeadersMessage) outbound(writeTarget);
-        List<Sha256Hash> expectedLocator = new ArrayList<>();
+        BlockLocator expectedLocator = new BlockLocator();
         expectedLocator.add(b1.getHash());
         expectedLocator.add(UNITTEST.getGenesisBlock().getHash());
-        assertEquals(getheaders.getLocator(), expectedLocator);
+        assertEquals(getheaders.getLocator().getHashes(), expectedLocator.getHashes());
         assertEquals(getheaders.getStopHash(), Sha256Hash.ZERO_HASH);
         // Now send all the headers.
         HeadersMessage headers = new HeadersMessage(UNITTEST, b2.cloneAsHeader(),
                 b3.cloneAsHeader(), b4.cloneAsHeader());
         // We expect to be asked for b3 and b4 again, but this time, with a body.
-        expectedLocator.clear();
+        expectedLocator = new BlockLocator();
         expectedLocator.add(b2.getHash());
         expectedLocator.add(b1.getHash());
         expectedLocator.add(UNITTEST.getGenesisBlock().getHash());
         inbound(writeTarget, headers);
         GetBlocksMessage getblocks = (GetBlocksMessage) outbound(writeTarget);
-        assertEquals(expectedLocator, getblocks.getLocator());
+        assertEquals(expectedLocator.getHashes(), getblocks.getLocator().getHashes());
```

Comment:
Consider implementing `BlockLocator.equals()` so you can compare two locators directly just like the lists were compared directly before.